src/workflow_components/composite_action.py

```diff
@@ -108,6 +113,11 @@ def from_dict(obj_dict) -> "CompositeAction":
         ca = CompositeAction(name=obj_dict.get("name"), path=obj_dict["path"])
```

Comment:
I think it would be better to add the tag/ref to be constructor parameter

```suggestion
        ca = CompositeAction(name=obj_dict.get("name"), path=obj_dict["path"], tag=obj_dict.get("tag"))
```

Reply:
Why? It is not always stated. 
Will you do the same for workflows? Since only a few if the reusable workflows have tags.

Reply:
I wrote that understanding that it is a ref, and is part of the composite aciton/workflow identification. Each fetched workflow had a ref, it was just hardcoded as main/master.
But if we not treating this as a ref, but only as a supplementary tag, so maybe we do not need to have this is the constructor.